chanzuckerberg/single-cell-explorer
# fix: SVG import

Changes:
1. Update webpack `file-loader` to look for all specified file formats in the tree instead of previously restricted `node_modules` and `fonts` folders. This way we can import SVG files anywhere in the project as a file path string
2. Create a basic SvgIcon `img` component to take attribute `src={file_path}`
3. Update `package.json` and `package-lock.json` with `"@material-ui/lab": "^4.0.0-alpha.61",`, otherwise `make build-for-server-dev` would complain about `package.json` and `package-lock.json` are not in sync
4. Remove `publicPath: ".."` in `client/configuration/webpack/webpack.config.prod.js`, since we no longer host our own font files, and we need relative path to load static files like SVG

```
{
        test: /\.(jpg|png|gif|eot|svg|ttf|woff|woff2|otf)$/i,
        loader: "file-loader",
        options: {
          name: "static/assets/[name]-[contenthash].[ext]",
          publicPath: "..", // <--- this is no longer needed
        },
      },
```

Proof that it still works:

<img width="717" alt="Screen Shot 2022-08-11 at 1 41 25 PM" src="https://user-images.githubusercontent.com/6309723/184238588-e7e8f352-4ba6-495a-99fa-dd8ca7639ace.png">

QA:

1. Run `make build-for-server-dev && make start-server` and check out `http://localhost:5005/d/pbmc3k.cxg/`. The SVG icon should work so is the rest of the app!
2. Run `make start-frontend` and check out `http://localhost:3000/d/pbmc3k.cxg`. Everything should work there too 👌
